inc/functions.php

```diff
@@ -549,6 +549,64 @@ function my_date($format, $stamp=0, $offset="", $ty=1, $adodb=false)
 	return $date;
 }
 
+/**
+ * Get a mail handler instance, a MyBB's built-in SMTP / PHP mail hander or one created by a plugin.
+ * @param bool $use_buitlin Whether to use MyBB's built-in mail handler.
+ *
+ * @return object A MyBB built-in mail handler or one created by plugin(s).
+ */
+function &get_my_mailhandler($use_buitlin = false)
+{
+	global $mybb, $plugins;
+	static $my_mailhandler;
+	static $my_mailhandler_builtin;
+
+	if($use_buitlin)
+	{
+		// If our built-in mail handler doesn't exist, create it.
+		if(!is_object($my_mailhandler_builtin))
+		{
+			require_once MYBB_ROOT . "inc/class_mailhandler.php";
+
+			// Using SMTP.
+			if(isset($mybb->settings['mail_handler']) && $mybb->settings['mail_handler'] == 'smtp')
+			{
+				require_once MYBB_ROOT . "inc/mailhandlers/smtp.php";
+				$my_mailhandler_builtin = new SmtpMail();
+			}
+			// Using PHP mail().
+			else
+			{
+				require_once MYBB_ROOT . "inc/mailhandlers/php.php";
+				$my_mailhandler_builtin = new PhpMail();
+				if(!empty($mybb->settings['mail_parameters']))
+				{
+					$my_mailhandler_builtin->additional_parameters = $mybb->settings['mail_parameters'];
+				}
+			}
```

Comment:
I won't be sure, but perhaps a hook here could be handy? Maybe not much.

Reply:
What's the purpose of this hook? We may add it later if we're not so sure about it at present time.

Reply:
Hmm, on a second thought, the two hooks below may be useful here as well. It was that in Euan's original PR.

Reply:
I think this assumes `get_my_mailhandler(true)` will never be used by third parties, which might not be the case.